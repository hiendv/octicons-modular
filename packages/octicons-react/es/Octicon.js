import PropTypes from 'prop-types';
import React from 'react';

var Octicon = function (ref) {
  var icon = ref.icon;
  var scale = ref.scale;
  var className = ref.className;
  var label = ref.label;

  var options = { scale: scale, class: className, label: label };
  var attrs = icon.attrs(options);
  if (!attrs) {
    return (null)
  }

  Object.keys(attrs).forEach(function (idx) {
    attrs[idx] = "" + (attrs[idx]);
  });

  attrs.className = attrs.class;
  delete attrs.class;

  var pathAttrs = icon.path();
  pathAttrs.key = 'p0';

  return React.createElement('svg', attrs, [ React.createElement('path', pathAttrs) ])
};

Octicon.propTypes = {
  icon: PropTypes.shape({
    attrs: PropTypes.func.isRequired,
    path: PropTypes.func.isRequired
  }),
  scale: PropTypes.number,
  className: PropTypes.string,
  label: PropTypes.string
};

Octicon.defaultProps = {
  scale: 1,
  className: null,
  label: null
};

export default Octicon;
