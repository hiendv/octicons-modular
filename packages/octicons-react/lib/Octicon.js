'use strict';

var PropTypes = require('prop-types');
var React = require('react');

var Octicon = function (ref) {
  var icon = ref.icon;
  var scale = ref.scale;
  var className = ref.className;
  var label = ref.label;

  var options = { scale: scale, class: className, label: label };
  var attrs = icon.attrs(options);
  if (!attrs) {
    return (null)
  }

  Object.keys(attrs).forEach(function (idx) {
    attrs[idx] = "" + (attrs[idx]);
  });

  attrs.className = attrs.class;
  delete attrs.class;

  var pathAttrs = icon.path();
  pathAttrs.key = 'p0';

  return React.createElement('svg', attrs, [ React.createElement('path', pathAttrs) ])
};

Octicon.propTypes = {
  icon: PropTypes.shape({
    attrs: PropTypes.func.isRequired,
    path: PropTypes.func.isRequired
  }),
  scale: PropTypes.number,
  className: PropTypes.string,
  label: PropTypes.string
};

Octicon.defaultProps = {
  scale: 1,
  className: null,
  label: null
};

module.exports = Octicon;
