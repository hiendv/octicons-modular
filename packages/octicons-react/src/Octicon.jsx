import PropTypes from 'prop-types'
import React from 'react'
import Octicons from 'octicons-modular'
import 'octicons-modular/lib/main.css'

const getAttrs = element => {
  return Array.from(element.attributes).reduce((o, item) => {
    o[item.name] = item.value
    return o
  }, {})
}

let Octicon = ({ icon, scale, className, label }) => {
  let octicon = icon.svg({ scale, class: className, label })
  if (!octicon) {
    return (null)
  }

  let attrs = getAttrs(octicon)
  attrs.dangerouslySetInnerHTML = {__html: octicon.innerHTML}
  attrs.className = attrs.class
  delete attrs.class

  return React.createElement(
    octicon.tagName,
    attrs
  )
}

Octicon.propTypes = {
  icon: PropTypes.shape({
    svg: PropTypes.func
  }),
  scale: PropTypes.number,
  className: PropTypes.string,
  label: PropTypes.string
}

Octicon.defaultProps = {
  icon: {
    svg () {}
  },
  scale: 1,
  className: null,
  label: null
}

export * from 'octicons-modular'
export { Octicons, Octicon }
export default Octicon
