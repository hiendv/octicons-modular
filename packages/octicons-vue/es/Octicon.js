var Octicon = {
  functional: true,
  props: {
    icon: {
      type: Object,
      required: true,
      validator: function validator (value) {
        return value.attrs instanceof Function && value.path instanceof Function
      }
    },
    scale: {
      type: Number,
      default: 1
    },
    className: {
      type: String,
      default: null
    },
    label: {
      type: String,
      default: null
    }
  },
  render: function render (createElement, ref) {
    var props = ref.props;

    var icon = props.icon;
    var scale = props.scale;
    var className = props.className;
    var label = props.label;
    var options = { scale: scale, class: className, label: label };

    return createElement('svg', { attrs: icon.attrs(options) }, [ createElement('path', { attrs: icon.path() }) ])
  }
};

export default Octicon;
