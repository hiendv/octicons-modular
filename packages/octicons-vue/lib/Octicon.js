'use strict';

var getAttrs = function (element) {
  return Array.from(element.attributes).reduce(function (o, item) {
    o[item.name] = item.value;
    return o
  }, {})
};

var Octicon = {
  functional: true,
  props: {
    icon: {
      type: Object,
      required: true,
      validator: function validator (value) {
        return value.svg instanceof Function
      }
    },
    scale: {
      type: Number,
      default: 1
    },
    className: {
      type: String,
      default: null
    },
    label: {
      type: String,
      default: null
    }
  },
  render: function render (createElement, ref) {
    var props = ref.props;

    var icon = props.icon;
    var scale = props.scale;
    var className = props.className;
    var label = props.label;
    var octicon = icon.svg({ scale: scale, class: className, label: label });
    if (!octicon) {
      return
    }

    var attrs = getAttrs(octicon);
    var innerHTML = octicon.innerHTML;

    return createElement(
      octicon.tagName,
      {
        attrs: attrs,
        domProps: {
          innerHTML: innerHTML
        }
      }
    )
  }
};

module.exports = Octicon;
