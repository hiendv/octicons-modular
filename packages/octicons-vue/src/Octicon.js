export default {
  functional: true,
  props: {
    icon: {
      type: Object,
      required: true,
      validator (value) {
        return value.svg instanceof Function
      }
    },
    scale: {
      type: Number,
      default: 1
    },
    className: {
      type: String,
      default: null
    },
    label: {
      type: String,
      default: null
    }
  },
  render (createElement, { props }) {
    let { icon, scale, className, label } = props
    let options = { scale, class: className, label }
    let octicon = icon.svg(options)
    if (!octicon) {
      return
    }

    let attrs = icon.attrs(options)
    let innerHTML = octicon.innerHTML

    return createElement(
      octicon.tagName,
      {
        attrs,
        domProps: {
          innerHTML
        }
      }
    )
  }
}
