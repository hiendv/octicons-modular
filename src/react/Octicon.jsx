import PropTypes from 'prop-types'
import React from 'react'

const Octicon = ({ icon, scale }) => {
  let div = document.createElement('div')
  div.innerHTML = icon.svg({ scale: scale })
  let svg = div.firstChild
  let attrs = {}

  for (let attr of svg.attributes) {
    if (attr.name === 'class') {
      attrs.className = attr.value
      continue
    }

    attrs[attr.name] = attr.value
  }

  attrs.dangerouslySetInnerHTML = { __html: svg.innerHTML }

  return React.createElement('svg', attrs)
}

Octicon.propTypes = {
  icon: PropTypes.shape({
    svg: PropTypes.func
  }),
  scale: PropTypes.number
}

Octicon.defaultProps = {
  icon: {
    svg () {}
  },
  scale: 1
}

export default Octicon
