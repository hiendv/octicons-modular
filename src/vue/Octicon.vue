<style lang="scss">
  @import '../style.scss';
</style>
<script>
export default {
  name: 'Octicon',
  props: {
    icon: {
      type: Object,
      default () {
        return {
          svg () {
            return '<svg/>'
          }
        }
      },
      validator (value) {
        return typeof value.svg === 'function'
      }
    },
    scale: {
      type: Number,
      default () {
        return 1
      }
    }
  },
  render (createElement) {
    let div = document.createElement('div')
    div.innerHTML = this.icon.svg({ scale: this.scale })
    let svg = div.firstChild
    let attrs = {}

    for (let attr of svg.attributes) {
      attrs[attr.name] = attr.value
    }

    return createElement('svg', {
      attrs,
      domProps: {
        innerHTML: svg.innerHTML
      }
    })
  }
}
</script>
