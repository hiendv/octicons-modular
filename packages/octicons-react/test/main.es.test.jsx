import React from 'react'
import Renderer from 'react-test-renderer'
import Octicon, { zap } from '../es/main.js'

describe('Octicon component', () => {
  test('invalid', () => {
    const component = Renderer.create(
      <Octicon icon={{ svg () {} }} />
    )

    let tree = component.toJSON()
    expect(tree).toBeFalsy()
  })

  test('valid', () => {
    const component = Renderer.create(
      <Octicon icon={zap} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('scalable', () => {
    const component = Renderer.create(
      <Octicon icon={zap} scale={2} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('scalable with 0', () => {
    const component = Renderer.create(
      <Octicon icon={zap} scale={0} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('scalable with fractional values', () => {
    const component = Renderer.create(
      <Octicon icon={zap} scale={1.2345} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('accepts an additional class', () => {
    const component = Renderer.create(
      <Octicon icon={zap} className="an-additional-class" />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('ignores invalid additional classes', () => {
    const component = Renderer.create(
      <Octicon icon={zap} className={null} />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('accepts an label', () => {
    const component = Renderer.create(
      <Octicon icon={zap} label="a label" />
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
