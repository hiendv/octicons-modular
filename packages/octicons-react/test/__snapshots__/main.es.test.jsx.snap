// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Octicon component accepts an additional class 1`] = `
<svg
  aria-hidden="true"
  className="octicon octicon-zap an-additional-class"
  height="16"
  version="1.1"
  viewBox="0 0 10 16"
  width="10"
>
  <path
    d="M10 7H6l3-7-9 9h4l-3 7 9-9z"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`Octicon component accepts an label 1`] = `
<svg
  aria-label="a label"
  className="octicon octicon-zap"
  height="16"
  version="1.1"
  viewBox="0 0 10 16"
  width="10"
>
  <path
    d="M10 7H6l3-7-9 9h4l-3 7 9-9z"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`Octicon component ignores invalid additional classes 1`] = `
<svg
  aria-hidden="true"
  className="octicon octicon-zap"
  height="16"
  version="1.1"
  viewBox="0 0 10 16"
  width="10"
>
  <path
    d="M10 7H6l3-7-9 9h4l-3 7 9-9z"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`Octicon component scalable 1`] = `
<svg
  aria-hidden="true"
  className="octicon octicon-zap"
  height="32"
  version="1.1"
  viewBox="0 0 10 16"
  width="20"
>
  <path
    d="M10 7H6l3-7-9 9h4l-3 7 9-9z"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`Octicon component scalable with 0 1`] = `
<svg
  aria-hidden="true"
  className="octicon octicon-zap"
  height="0"
  version="1.1"
  viewBox="0 0 10 16"
  width="0"
>
  <path
    d="M10 7H6l3-7-9 9h4l-3 7 9-9z"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`Octicon component scalable with fractional values 1`] = `
<svg
  aria-hidden="true"
  className="octicon octicon-zap"
  height="19.75"
  version="1.1"
  viewBox="0 0 10 16"
  width="12.34"
>
  <path
    d="M10 7H6l3-7-9 9h4l-3 7 9-9z"
    fill-rule="evenodd"
  />
</svg>
`;

exports[`Octicon component valid 1`] = `
<svg
  aria-hidden="true"
  className="octicon octicon-zap"
  height="16"
  version="1.1"
  viewBox="0 0 10 16"
  width="10"
>
  <path
    d="M10 7H6l3-7-9 9h4l-3 7 9-9z"
    fill-rule="evenodd"
  />
</svg>
`;
